<!DOCTYPE html>
<html lang="en"><head>
<script src="day_5_files/libs/clipboard/clipboard.min.js"></script>
<script src="day_5_files/libs/quarto-html/tabby.min.js"></script>
<script src="day_5_files/libs/quarto-html/popper.min.js"></script>
<script src="day_5_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="day_5_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="day_5_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="day_5_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="day_5_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.4.550">

  <title>Day 5 - Single-cell and Spatial Transcriptomics</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="day_5_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="day_5_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
  </style>
  <link rel="stylesheet" href="day_5_files/libs/revealjs/dist/theme/quarto.css">
  <link href="day_5_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="day_5_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="day_5_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="day_5_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    padding-bottom: 0.5rem;
    margin-bottom: 0;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Day 5 - Single-cell and Spatial Transcriptomics</h1>
  <p class="subtitle">Sequencing one cell at a time</p>

<div class="quarto-title-authors">
</div>

</section>
<section id="course-topics" class="slide level2">
<h2>Course Topics</h2>
<ol type="1">
<li class="fragment"><strong>Day 1:</strong> Brief Introduction + Quantitative RNA methods;</li>
<li class="fragment"><strong>Day 2:</strong> Microarray: First Genome-Wide View;</li>
<li class="fragment"><strong>Day 3:</strong> RNAseq: Short-Read Revolution;</li>
<li class="fragment"><strong>Day 4:</strong> Long-Read Transcriptomics; (To be scheduled)</li>
<li class="fragment"><strong>Day 5: Single-cell and Spatial Transcriptomics;</strong></li>
</ol>
</section>
<section id="yesterday-summary" class="slide level2 smaller">
<h2>“Yesterday” summary</h2>
<table>
<colgroup>
<col style="width: 11%">
<col style="width: 12%">
<col style="width: 20%">
<col style="width: 29%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th>Platform</th>
<th>Detection</th>
<th>Base ID Method</th>
<th>Key Feature</th>
<th style="text-align: right;">Est. Cost (30M PE reads, 2×150 bp)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>Illumina</strong></td>
<td>Fluorescence</td>
<td>4-color reversible terminators</td>
<td>High accuracy, dominant in RNA-seq</td>
<td style="text-align: right;">~$90–$120 per sample <em>(core facility)</em></td>
</tr>
<tr class="even">
<td><strong>BGI DNBSEQ</strong></td>
<td>Fluorescence</td>
<td>cPAS chemistry</td>
<td>Cost-efficient, comparable accuracy</td>
<td style="text-align: right;">~$25–$60 per sample <em>(via service quote)</em></td>
</tr>
<tr class="odd">
<td><strong>Ion Torrent</strong></td>
<td>pH (electrical)</td>
<td>Semiconductor detection</td>
<td>No fluorescence, faster but noisier</td>
<td style="text-align: right;">~$400–$2,000 per sample <em>(depends on run type)</em></td>
</tr>
</tbody>
</table>
<blockquote>
<p>💡 <em>Estimates based on reagent + core facility pricing (9 Gb ≈ 30 M paired-end reads). Actual prices depend on run configuration, multiplexing, and region.</em></p>
</blockquote>
</section>
<section id="todays-goals" class="slide level2 smaller">
<h2>Today’s Goals</h2>
<p>By the end of today, you should be able to:</p>
<ol type="1">
<li class="fragment"><strong>Why move to single-cell resolution?</strong></li>
<li class="fragment"><strong>Overview of scRNA-seq methods</strong>
<ul>
<li class="fragment">Dissociation</li>
<li class="fragment">Isolation;</li>
<li class="fragment">Cell lysis / RT;</li>
<li class="fragment">Amplification;</li>
<li class="fragment">Library preparation;</li>
</ul></li>
<li class="fragment"><strong>Key computational concepts</strong>
<ul>
<li class="fragment">Quality control, Normalization, dimensionality reduction, clustering, etc</li>
</ul></li>
<li class="fragment"><strong>Spatial transcriptomics technologies</strong>
<ul>
<li class="fragment">Visium / Slide-Seq</li>
</ul></li>
</ol>
</section>
<section class="slide level2">

</section>
<section>
<section id="why-move-to-single-cell-resolution" class="title-slide slide level1 center">
<h1>Why move to single-cell resolution?</h1>

</section>
<section id="bulk-vs.-rnaseq" class="slide level2">
<h2>Bulk vs.&nbsp;RNAseq</h2>

<img data-src="https://www.lexogen.com/wp-content/uploads/2024/01/Difference-between-scRNA-Seq-and-bulk-RNA-Seq.png" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="evolution-of-scrnaseq" class="slide level2">
<h2>Evolution of scRNAseq</h2>

<img data-src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fnprot.2017.149/MediaObjects/41596_2018_Article_BFnprot2017149_Fig1_HTML.jpg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="evolution-of-scrnaseq-1" class="slide level2">
<h2>Evolution of scRNAseq</h2>

<img data-src="https://encyclopedia.pub/media/item_content/202206/62bc11298d210biosensors-12-00450-g002.png" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="droplet-based" class="slide level2">
<h2>Droplet-based</h2>

<img data-src="https://i0.wp.com/1.bp.blogspot.com/-jwTaEJV7B4k/VV8XinPfqzI/AAAAAAAABpw/IqdCfYCsNmE/s1600/Dropseq1.jpg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="droplet-based-1" class="slide level2">
<h2>Droplet-based</h2>
<p><br> <br></p>

<img data-src="https://mccarrolllab.org/wp-content/uploads/2015/05/Substack-4455-4504.gif" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="split-seq" class="slide level2">
<h2>SPLiT-seq</h2>

<img data-src="https://cdn.ncbi.nlm.nih.gov/pmc/blobs/a6ae/7643870/0fe6c5496ae9/nihms-1581077-f0001.jpg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="x-genomics" class="slide level2">
<h2>10x Genomics</h2>
<p><br></p>

<img data-src="https://i2.wp.com/2.bp.blogspot.com/-lkcFwl7hu2U/V5oT2lyrdbI/AAAAAAAACSI/p72Uv26K9BkKhvkEVdHb2FcD3Ts7KFqqgCLcB/s1600/Screen%2BShot%2B2016-07-28%2Bat%2B15.16.35.png?ssl=1" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="questions-could-be-answered" class="slide level2 smaller">
<h2>Questions could be answered</h2>
<ol type="1">
<li class="fragment">Which genetic markers are expressed among different cell types?</li>
<li class="fragment">What is the composition of cells in my sample? New or rare subpopulations of cells?</li>
<li class="fragment">What activation or differentiation states exist within a population?</li>
<li class="fragment">Does the cell composition change between conditions (healthy vs.&nbsp;diseased)?</li>
<li class="fragment">How do cells change over time or in response to stimuli?</li>
<li class="fragment">How does the cellular microenvironment influence a cell’s function?</li>
</ol>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="overview-of-scrna-seq-methods" class="title-slide slide level1 center">
<h1>Overview of scRNA-seq methods</h1>

</section>
<section id="basic-workflow-10x-genomics" class="slide level2">
<h2>Basic workflow (10x Genomics)</h2>
<ol type="1">
<li class="fragment">Dissociation;</li>
<li class="fragment">Isolation;**</li>
<li class="fragment">Cell lysis / RT;**</li>
<li class="fragment">Amplification;</li>
<li class="fragment">Library preparation;</li>
<li class="fragment">Sequencing;</li>
<li class="fragment">Data analysis;</li>
</ol>
<blockquote>
<p>**Key steps of scRNAseq protocol.</p>
</blockquote>
</section>
<section id="basic-workflow-10x-genomics-1" class="slide level2">
<h2>Basic workflow (10x Genomics)</h2>
<p><br></p>

<img data-src="https://encyclopedia.pub/media/item_content/202206/62bc1112b6b12biosensors-12-00450-g001.png" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="dissociation" class="slide level2">
<h2>Dissociation</h2>

<img data-src="https://media.springernature.com/lw685/springer-static/image/chp%3A10.1007%2F978-1-0716-3995-5_19/MediaObjects/535446_2_En_19_Fig5_HTML.png" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"><blockquote>
<p>Viability &gt;85% is required for good-quality libraries.</p>
</blockquote>
</section>
<section id="isolation" class="slide level2">
<h2>Isolation</h2>
<p><br></p>

<img data-src="https://cdn.10xgenomics.com/image/upload/v1709930681/blog/GEM-X%20Launch%20blog/Figure_1.png" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="cell-lysis-rt" class="slide level2">
<h2>Cell lysis / RT</h2>
<p><br> <br></p>

<img data-src="https://cdn.10xgenomics.com/image/upload/v1710226015/support/3_Gel_Bead.png" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="amplification" class="slide level2">
<h2>Amplification</h2>

<img data-src="https://hbctraining.github.io/scRNA-seq/img/umis.png" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="library-preparation" class="slide level2">
<h2>Library preparation</h2>
<p><br> <br></p>

<img data-src="https://cdn.10xgenomics.com/image/upload/v1755542613/knowledge-base/4404050374925.png" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="bias-of-distribution" class="slide level2">
<h2>Bias of distribution</h2>

<img data-src="https://www.researchgate.net/publication/350206220/figure/fig4/AS:1003487005523969@1616261354129/Fig-S1-A-3-biased-distribution-of-10x-Chromium-scRNA-seq-on-long-noncoding-RNA.jpg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="scrnaseq-workflow" class="slide level2">
<h2>scRNAseq Workflow</h2>
<div style="display: flex; justify-content: center; align-items: center; height: 70vh;">
<p><iframe src="https://www.youtube.com/embed/watch?v=eVEDw1YykME" title="Illumina sequencing" style="width: 80vw; height: 45vw; max-width: 960px; max-height: 540px; border: none;" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""> </iframe></p>
</div>
</section>
<section id="how-much-depth-and-reads-per-cell" class="slide level2">
<h2>How much depth and reads per cell?</h2>
<p>The required depth for RNA-seq depends on the study’s goal, but a common range is <strong>30-60 million reads per sample</strong> for a broad view of gene expression. If you are managing to sequence 10k cells, you will probably end up with <strong>2000-3500 read pairs per cell</strong>.</p>
<blockquote>
<p>Not a precise equation, depends on which experiment you are designing.</p>
</blockquote>
</section>
<section id="how-many-cells-should-be-sequenced" class="slide level2">
<h2>How many cells should be sequenced?</h2>
<p>https://satijalab.org/howmanycells/</p>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="key-computational-concepts" class="title-slide slide level1 center">
<h1>Key computational concepts</h1>

</section>
<section id="pre-processing" class="slide level2">
<h2>Pre-processing</h2>
<p>Usage of <strong>Cell Ranger</strong> — the standard preprocessing software provided by <strong>10x Genomics</strong> — is the first computational step in analyzing raw single-cell RNA-seq data.<br>
It transforms <strong>Illumina BCL or FASTQ files</strong> into a <strong>gene-cell expression matrix</strong>, which serves as the foundation for downstream analyses.</p>
</section>
<section id="main-steps" class="slide level2">
<h2>Main steps</h2>
<ol type="1">
<li class="fragment">Quality control</li>
<li class="fragment">Normalization</li>
<li class="fragment">Feature selection</li>
<li class="fragment">Dimension reduction</li>
<li class="fragment">Cell annotation</li>
<li class="fragment">Downstream analysis</li>
</ol>
</section>
<section id="main-steps-1" class="slide level2">
<h2>Main steps</h2>

<img data-src="https://media.springernature.com/lw1200/springer-static/image/art%3A10.1038%2Fs41576-023-00586-w/MediaObjects/41576_2023_586_Fig2_HTML.png" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="quality-control" class="slide level2">
<h2>Quality control</h2>

<img data-src="https://www.sc-best-practices.org/_images/quality_control.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="normalization" class="slide level2">
<h2>Normalization</h2>
<p>The preprocessing step of “normalization” aims to adjust the raw counts in the dataset for variable sampling effects by <strong>scaling the observable variance to a specified range</strong>. Several normalization techniques are used in practice varying in complexity. They are mostly designed in such a way that subsequent analysis tasks and their underlying statistical methods are applicable.</p>
</section>
<section id="feature-selection" class="slide level2">
<h2>Feature selection</h2>
<p><br> <br></p>

<img data-src="https://www.sc-best-practices.org/_images/feature_selection.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="dimension-reduction" class="slide level2">
<h2>Dimension reduction</h2>
<p><br> <br></p>

<img data-src="https://www.sc-best-practices.org/_images/dimensionality_reduction.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="cell-annotation" class="slide level2">
<h2>Cell annotation</h2>

<img data-src="https://satijalab.org/seurat/articles/pbmc3k_tutorial_files/figure-html/umapplot-1.png" class="quarto-figure quarto-figure-center r-stretch"></section>
<section id="cell-annotation-1" class="slide level2">
<h2>Cell annotation</h2>

<img data-src="https://satijalab.org/seurat/articles/pbmc3k_tutorial_files/figure-html/unnamed-chunk-3-1.png" class="quarto-figure quarto-figure-center r-stretch"></section>
<section id="cell-annotation-2" class="slide level2">
<h2>Cell annotation</h2>

<img data-src="https://satijalab.org/seurat/articles/pbmc3k_tutorial_files/figure-html/labelplot-1.png" class="quarto-figure quarto-figure-center r-stretch"></section>
<section id="downstream-analysis" class="slide level2">
<h2>Downstream analysis</h2>

<img data-src="https://www.sc-best-practices.org/_images/differential_gene_expression.jpg" class="quarto-figure quarto-figure-center r-stretch"></section>
<section id="downstream-analysis-1" class="slide level2">
<h2>Downstream analysis</h2>

<img data-src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41467-020-14766-3/MediaObjects/41467_2020_14766_Fig6_HTML.png" class="quarto-figure quarto-figure-center r-stretch"></section>
<section id="scrnaseq-book" class="slide level2">
<h2>scRNAseq book</h2>

<img data-src="https://www.sc-best-practices.org/_images/title.png" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"><blockquote>
<p>https://www.sc-best-practices.org/preamble.html</p>
</blockquote>
</section>
<section id="fictional-design-request-1" class="slide level2">
<h2>Fictional design request #1</h2>
<p>You are working on a project that is trying to describe function of a KO in a tissue. You have 4 vs.&nbsp;4 samples for each condition. You are working with a tissue with 7 known cell types. Your rarest cell type is expected to be 2% and you want to achieve at least 200 cells in all cell types.</p>
<ol type="1">
<li class="fragment">How much depth should be sequenced per sample? Read pairs per cell?</li>
<li class="fragment">How many cells per samples you are aiming?</li>
</ol>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="spatial-transcriptomics-technologies" class="title-slide slide level1 center">
<h1>Spatial transcriptomics technologies</h1>

</section>
<section id="what-is-it-about" class="slide level2 smaller">
<h2>What is it about?</h2>
<p>Spatial transcriptomics allows researchers to <strong>quantify gene expression while preserving spatial context</strong> within a tissue section.</p>
<ul>
<li class="fragment">Each measurement is linked to a precise <strong>location on the tissue</strong>.<br>
</li>
<li class="fragment">Combines <strong>histological imaging</strong> with <strong>RNA sequencing</strong> to map cellular functions in space.<br>
</li>
<li class="fragment">Enables discovery of:
<ul>
<li class="fragment">Tissue organization and cell interactions<br>
</li>
<li class="fragment">Spatial gradients in gene expression<br>
</li>
<li class="fragment">Regional responses to disease or injury</li>
</ul></li>
</ul>
<p>🧠 <em>Think of it as scRNA-seq + coordinates → a map of molecular activity within tissues.</em></p>
</section>
<section id="overview" class="slide level2 smaller">
<h2>Overview</h2>

<img data-src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41586-021-03634-9/MediaObjects/41586_2021_3634_Fig1_HTML.png?as=webp" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="visium-10x-genomics" class="slide level2 smaller">
<h2>Visium (10x Genomics)</h2>
<p><strong>10x Genomics Visium</strong> is one of the most widely used spatial transcriptomics platforms.</p>
<ul>
<li class="fragment">Uses glass slides covered with <strong>55 μm barcoded capture spots</strong>, each containing millions of oligo-dT probes.<br>
</li>
<li class="fragment">Captures mRNA directly from tissue placed on the slide, maintaining <strong>spatial information</strong> via spot-specific barcodes.<br>
</li>
<li class="fragment">Generates paired data:
<ul>
<li class="fragment"><strong>Histological image</strong> (H&amp;E stain)<br>
</li>
<li class="fragment"><strong>Spatial gene expression matrix</strong></li>
</ul></li>
</ul>
<p>🔬 Resolution: ~1–10 cells per spot<br>
📊 Integration: Compatible with <strong>Seurat</strong>, <strong>Scanpy</strong>, and <strong>Space Ranger</strong> pipelines.</p>
</section>
<section id="visium-10x-genomics-1" class="slide level2 smaller">
<h2>Visium (10x Genomics)</h2>
<p><br> <br></p>

<img data-src="https://cdn.10xgenomics.com/image/upload/v1645041033/analysis-guides/Spatial-AnnotatedSCdata-Illus.png" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="slide-seq" class="slide level2 smaller">
<h2>Slide-seq</h2>
<p><strong>Slide-seq</strong> achieves higher spatial resolution using <strong>DNA-barcoded beads</strong>.</p>
<ul>
<li class="fragment">A surface is coated with thousands of <strong>10 µm beads</strong>, each with a unique positional barcode.<br>
</li>
<li class="fragment">Tissue is transferred onto the bead array; mRNA diffuses and binds to nearby beads.<br>
</li>
<li class="fragment">After sequencing, spatial positions are reconstructed using the known bead coordinates.<br>
</li>
<li class="fragment">Enables near <strong>single-cell resolution</strong> across entire tissue sections.</li>
</ul>
</section>
<section id="slide-seq-1" class="slide level2 smaller">
<h2>Slide-seq</h2>
<p><br></p>

<img data-src="https://www.science.org/cms/10.1126/science.aaw1219/asset/c2b8070a-1bfb-4c2c-9c75-aedb8143dbfd/assets/graphic/363_1463_f1.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="platforms" class="slide level2 smaller">
<h2>Platforms</h2>
<table>
<colgroup>
<col style="width: 31%">
<col style="width: 20%">
<col style="width: 23%">
<col style="width: 23%">
</colgroup>
<thead>
<tr class="header">
<th>Platform / Method</th>
<th>Resolution</th>
<th>Capture Type</th>
<th>Key Features</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>10x Genomics Visium</strong></td>
<td>55 µm spot</td>
<td>PolyA capture</td>
<td>Combines imaging with transcriptomics; widely adopted</td>
</tr>
<tr class="even">
<td><strong>Slide-seq / Slide-seqV2</strong></td>
<td>10 µm bead</td>
<td>PolyA capture</td>
<td>High spatial resolution using barcoded bead arrays</td>
</tr>
<tr class="odd">
<td><strong>MERFISH / seqFISH</strong></td>
<td>Single-molecule</td>
<td>Imaging-based</td>
<td>Detects hundreds–thousands of genes <em>in situ</em></td>
</tr>
</tbody>
</table>
</section>
<section id="break" class="slide level2">
<h2>Break</h2>
<p>☕ 15 min coffee break<br>
Stretch, chat, and prepare for the next section:<br>
<strong>Application &amp; Case Study</strong></p>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="application-case-study" class="title-slide slide level1 center">
<h1>Application &amp; Case study</h1>

</section>
<section id="today-assay" class="slide level2 smaller">
<h2>Today assay</h2>
<ul>
<li class="fragment">Research for ~1h for a RNAseq experiment in any research study related to single-cell RNAseq (avoid bulk-only RNAseq). Make a two slides presentation (5 minutes or less) in our colective google presentations:</li>
</ul>
<p>https://docs.google.com/presentation/d/1VA6orV_DraHiUIpzQW79o-O25CwcCFspmx00JYgDPD4/edit?usp=sharing</p>
<ul>
<li class="fragment">Answer these questions:
<ul>
<li class="fragment">How many cell it has in total?</li>
<li class="fragment">Based on https://satijalab.org/howmanycells/</li>
<li class="fragment">Single-end or paired-end? Read length?</li>
<li class="fragment">Which genome reference was used? If suitable, which annotation reference?</li>
<li class="fragment">Does it follow ENCODE best principles?</li>
</ul></li>
</ul>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="questions" class="title-slide slide level1 center">
<h1>Questions</h1>

</section>
<section id="q1-primary-goal-of-rna-seq" class="slide level2">
<h2>Q1: Primary Goal of RNA-seq</h2>
<p>What is the primary goal of the RNA-seq workflow?</p>
<ol type="A">
<li class="fragment">To physically sequence the entire length of the human genome.</li>
<li class="fragment">To quantify and compare transcript abundance across different samples.</li>
<li class="fragment">To align short reads to a Protein-Protein Interaction (PPI) network.</li>
<li class="fragment">To reconstruct a cDNA library from a FASTA file.</li>
</ol>
<div class="quarto-auto-generated-content">
<div class="footer footer-default">

</div>
</div>
</section></section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="day_5_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="day_5_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="day_5_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="day_5_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="day_5_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="day_5_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="day_5_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="day_5_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="day_5_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="day_5_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'fade',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        text: function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
        const config = {
          allowHTML: true,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start',
        };
        if (contentFn) {
          config.content = contentFn;
        }
        if (onTriggerFn) {
          config.onTrigger = onTriggerFn;
        }
        if (onUntriggerFn) {
          config.onUntrigger = onUntriggerFn;
        }
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>