<!DOCTYPE html>
<html lang="en"><head>
<script src="day_3_files/libs/clipboard/clipboard.min.js"></script>
<script src="day_3_files/libs/quarto-html/tabby.min.js"></script>
<script src="day_3_files/libs/quarto-html/popper.min.js"></script>
<script src="day_3_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="day_3_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="day_3_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="day_3_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="day_3_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.4.550">

  <title>Day 3 - NGS RNAseq</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="day_3_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="day_3_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
  </style>
  <link rel="stylesheet" href="day_3_files/libs/revealjs/dist/theme/quarto.css">
  <link href="day_3_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="day_3_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="day_3_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="day_3_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    padding-bottom: 0.5rem;
    margin-bottom: 0;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Day 3 - NGS RNAseq</h1>
  <p class="subtitle">Short-Read Revolution</p>

<div class="quarto-title-authors">
</div>

</section>
<section id="course-topics" class="slide level2">
<h2>Course Topics</h2>
<ol type="1">
<li class="fragment"><strong>Day 1:</strong> Brief Introduction + Quantitative RNA methods;</li>
<li class="fragment"><strong>Day 2:</strong> Microarray: First Genome-Wide View;</li>
<li class="fragment"><strong>Day 3: RNAseq: Short-Read Revolution;</strong></li>
</ol>
</section>
<section id="yesterday-summary" class="slide level2">
<h2>Yesterday summary</h2>
<div class="cell">
<div class="cell-output-display">
<table>
<thead>
<tr class="header">
<th style="text-align: left;">Platform</th>
<th style="text-align: center;">Throughput</th>
<th style="text-align: center;">Average_Price</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Affymetrix</td>
<td style="text-align: center;">30,000‚Äì50,000 probes/array</td>
<td style="text-align: center;">~$250‚Äì$400 per array</td>
</tr>
<tr class="even">
<td style="text-align: left;">Agilent</td>
<td style="text-align: center;">40,000‚Äì60,000 probes/array</td>
<td style="text-align: center;">~$200‚Äì$300 per array</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Illumina</td>
<td style="text-align: center;">&gt;48,000 probes/array</td>
<td style="text-align: center;">~$150‚Äì$250 per array</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Microarrays allow <strong>simultaneous measurement of thousands of RNA transcripts</strong> by hybridizing labeled cDNA to probes fixed on a solid surface.</p>
</section>
<section id="todays-goals" class="slide level2 smaller">
<h2>Today‚Äôs Goals</h2>
<p>By the end of today, you should be able to:</p>
<ol type="1">
<li class="fragment"><strong>RNA-seq short-read workflow overview</strong>
<ul>
<li class="fragment">Library prep, sequencing, and read mapping principles.</li>
</ul></li>
<li class="fragment"><strong>Experiment design in short-read RNAseq</strong>
<ul>
<li class="fragment">Replication, depth, batch control.</li>
</ul></li>
<li class="fragment"><strong>Core analysis logic</strong>
<ul>
<li class="fragment">Quality assessment ‚Üí alignment ‚Üí quantification</li>
<li class="fragment">FASTQ, BAM/SAM, count matrices.</li>
</ul></li>
<li class="fragment"><strong>Downstream analysis</strong>
<ul>
<li class="fragment">PCA, volcano plots, enrichment (conceptual level).</li>
</ul></li>
</ol>
</section>
<section class="slide level2">

</section>
<section>
<section id="rna-seq-workflow-overview" class="title-slide slide level1 center">
<h1>RNA-seq workflow overview</h1>

</section>
<section id="rna-seq-workflow" class="slide level2">
<h2>RNA-seq Workflow</h2>
<ul>
<li class="fragment"><strong>Goal:</strong> Quantify and compare transcript abundance across samples.<br>
</li>
<li class="fragment"><strong>Main steps:</strong>
<ol type="1">
<li class="fragment">RNA isolation**</li>
<li class="fragment">Library preparation</li>
<li class="fragment">Sequencing</li>
<li class="fragment">Read mapping &amp; quantification<br>
</li>
<li class="fragment">Downstream analysis (DE, splicing, pathways)</li>
</ol></li>
</ul>

<img data-src="https://upload.wikimedia.org/wikipedia/commons/6/6a/RNA_seq_workflow.svg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="library-preparation" class="slide level2 smaller">
<h2>Library Preparation</h2>
<ul>
<li class="fragment">RNA is first converted into <strong>cDNA fragments</strong>, typically 200‚Äì500 bp long.</li>
<li class="fragment"><strong>Adapters</strong> are ligated to both ends ‚Äî these are short, known sequences required for:
<ul>
<li class="fragment">Binding to the flow cell surface.</li>
<li class="fragment">Serving as priming sites for sequencing.</li>
</ul></li>
<li class="fragment">Optional steps include:
<ul>
<li class="fragment"><strong>Fragmentation</strong> (enzymatic or mechanical).</li>
<li class="fragment"><strong>Size selection</strong> for desired insert length.</li>
<li class="fragment"><strong>PCR amplification</strong> to enrich for adapter-ligated fragments.</li>
</ul></li>
</ul>
<blockquote>
<p>üéØ Goal: produce a high-quality cDNA library compatible with the sequencing platform.</p>
</blockquote>
</section>
<section id="common-elements" class="slide level2">
<h2>Common Elements</h2>
<ul>
<li class="fragment"><p><strong>Adapters</strong> ‚Äî Short, known DNA sequences ligated to fragment ends.<br>
Allow fragments to bind to the flow cell surface or nanoball array.</p></li>
<li class="fragment"><p><strong>Indexes / Barcodes</strong> ‚Äî Short identifier sequences embedded within adapters.<br>
Enable <strong>multiplexing</strong>, allowing several samples to be sequenced in one run.</p></li>
<li class="fragment"><p><strong>Priming Sites</strong> ‚Äî Regions within adapters where sequencing primers bind during each read.</p></li>
</ul>
</section>
<section id="common-elements-1" class="slide level2">
<h2>Common Elements</h2>

<img data-src="https://seqwell.com/wp-content/uploads/2023/03/barcode-1200x-1.png" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="cdna-fragment-illumina" class="slide level2">
<h2>cDNA fragment (Illumina)</h2>

<img data-src="https://knowledge.illumina.com/~gitbook/image?url=https%3A%2F%2F761066130-files.gitbook.io%2F%7E%2Ffiles%2Fv0%2Fb%2Fgitbook-x-prod.appspot.com%2Fo%2Fspaces%252FGM9W2DuBTgEXv1ClCm8H%252Fuploads%252Fgit-blob-d9099352b312adb3c3f15560cfab0744d9564ffa%252Fimage1.png%3Falt%3Dmedia%26token%3D2d294364-12d7-427e-9981-9948442f6b63&amp;width=768&amp;dpr=4&amp;quality=100&amp;sign=314070f7&amp;sv=2" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="sequencing-illumina" class="slide level2">
<h2>Sequencing (Illumina)</h2>
<div style="display: flex; justify-content: center; align-items: center; height: 70vh;">
<p><iframe src="https://www.youtube.com/embed/EDVKxSNdSic" title="Illumina sequencing" style="width: 80vw; height: 45vw; max-width: 960px; max-height: 540px; border: none;" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""> </iframe></p>
</div>
</section>
<section id="platforms" class="slide level2 smaller">
<h2>Platforms</h2>
<table>
<colgroup>
<col style="width: 18%">
<col style="width: 30%">
<col style="width: 27%">
<col style="width: 23%">
</colgroup>
<thead>
<tr class="header">
<th>Platform</th>
<th>Detection Signal</th>
<th>Base ID Method</th>
<th>Key Feature</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Illumina</td>
<td>Fluorescence</td>
<td>4-color reversible terminators</td>
<td>High accuracy, dominant in RNA-seq</td>
</tr>
<tr class="even">
<td>BGI DNBSEQ</td>
<td>Fluorescence</td>
<td>cPAS chemistry</td>
<td>Cost-efficient, similar accuracy</td>
</tr>
<tr class="odd">
<td>Ion Torrent</td>
<td>pH (electrical)</td>
<td>Semiconductor detection</td>
<td>No fluorescence, faster but noisier</td>
</tr>
</tbody>
</table>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="experiment-design-in-short-read-rnaseq" class="title-slide slide level1 center">
<h1>Experiment design in short-read RNAseq</h1>

</section>
<section id="encode-standards" class="slide level2">
<h2>ENCODE standards</h2>
<p>The ENCODE Project (Encyclopedia of DNA Elements) is a public research project that aims to build a comprehensive parts list of functional elements in the human genome. The ENCODE Data Coordination Center (DCC)‚Äôs primary task is to curate, uniformly process and validate the data generated and submitted by ENCODE Consortium members in preparation for release to the scientific community.</p>
<blockquote>
<p>https://www.encodeproject.org/data-standards/</p>
</blockquote>
</section>
<section id="relevant-questions" class="slide level2">
<h2>Relevant questions</h2>
<ol type="1">
<li class="fragment">How many biological replicates?</li>
<li class="fragment">How much sequencing depth?</li>
<li class="fragment">Read length</li>
<li class="fragment">Single vs.&nbsp;Paired end</li>
<li class="fragment">Stranded vs.&nbsp;unstranded</li>
</ol>
</section>
<section id="how-many-biological-replicates" class="slide level2">
<h2>How many biological replicates?</h2>
<p>A 48-replicate yeast study showed that many of the tools available for DGE analysis detected only 20‚Äì40% of differentially expressed genes when only 3 replicates were included in the analysis. The study suggested that a minimum of <strong>six biological replicates should be used</strong>, which is substantially more than the three or four replicates generally reported in the RNA-‚Äãseq literature.</p>
<blockquote>
<p>ENCODE4: Experiments should have two or more replicates.</p>
</blockquote>
</section>
<section id="how-much-sequencing-depth" class="slide level2">
<h2>How much sequencing depth?</h2>
<p>For bulk RNA DGE experiments in eukaryotic genomes, it is generally accepted that read depths of around 10‚Äì30 million reads per sample are required.</p>
<blockquote>
<p>ENCODE4: Each replicate should have <strong>30 million aligned reads</strong>, although older projects aimed for 20 million reads. Best practices for ENCODE2 RNA-seq experiments have been outlined .</p>
</blockquote>
</section>
<section id="read-length" class="slide level2">
<h2>Read length</h2>
<p>Read length determines how much of each transcript is sequenced and directly affects alignment accuracy and transcriptome coverage. Short reads (e.g., 50 bp) are sufficient for gene-level expression quantification, but longer reads (100‚Äì150 bp) improve mapping across splice junctions and enable better detection of alternative splicing and novel isoforms.</p>
</section>
<section id="single-vs.-paired-end" class="slide level2">
<h2>Single vs.&nbsp;Paired end</h2>
<p>Single-end sequencing reads one end of each cDNA fragment, while paired-end sequencing reads both ends, producing two reads per fragment.</p>
<p>Paired-end data provide improved mapping confidence, especially for repetitive regions, fusion gene detection, and alternative splicing analysis. However, paired-end runs are more expensive and produce larger data volumes.</p>
<p>Paired-end (e.g., 2 √ó 75 bp or 2 √ó 100 bp) is preferred for transcript discovery and isoform-level analysis.</p>
</section>
<section id="single-vs.-paired-end-1" class="slide level2">
<h2>Single vs.&nbsp;Paired end</h2>

<img data-src="https://statomics.github.io/SGA/images_sequencing/singleEndVsPairedEnd.png" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="stranded-vs.-unstranded" class="slide level2">
<h2>Stranded vs.&nbsp;unstranded</h2>
<p>Stranded (or directional) RNA-seq preserves the information about which DNA strand was transcribed, enabling accurate detection of antisense transcription and overlapping genes. Unstranded protocols lose this information, which can cause ambiguity in complex loci.</p>
<blockquote>
<p>ENCODE4: Recommends strand-specific libraries for all RNA-seq experiments to accurately quantify sense and antisense transcripts.</p>
</blockquote>
</section>
<section id="stranded-vs.-unstranded-1" class="slide level2">
<h2>Stranded vs.&nbsp;unstranded</h2>

<img data-src="https://biocorecrg.github.io/PHINDaccess_RNAseq_2020/images/rnaseq_strandedness.png" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="strandedness-vs.-dge-paradigm" class="slide level2">
<h2>Strandedness vs.&nbsp;DGE paradigm</h2>

<img data-src="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2Fs12864-015-1876-7/MediaObjects/12864_2015_1876_Fig4_HTML.gif?as=webp" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section class="slide level2">

</section></section>
<section>
<section id="core-analysis-logic" class="title-slide slide level1 center">
<h1>Core analysis logic</h1>

</section>
<section id="raw-data-fastq" class="slide level2 smaller">
<h2>Raw Data (FASTQ)</h2>
<p>Generally, we are speaking of <strong>FASTQ</strong> files. FASTQ is the raw data format produced by sequencing machines. Each read has four lines:</p>
<ol type="1">
<li class="fragment">Identifier (e.g., <span class="citation" data-cites="SEQ_ID">@SEQ_ID</span>)</li>
<li class="fragment">Nucleotide sequence</li>
<li class="fragment">Separator (+)</li>
<li class="fragment">Quality scores (ASCII-encoded Phred scores)</li>
</ol>
<p>FASTQ stores both the read sequences and their base quality, essential for downstream filtering and alignment.</p>
<blockquote>
<p>‚ÄúOfficial‚Äù raw data is BCL files that stands for <strong>binary base calling</strong> file format.</p>
</blockquote>
</section>
<section id="fastq-format" class="slide level2">
<h2>FASTQ format</h2>

<img data-src="https://biocorecrg.github.io/PhD_course_genomics_format_2021/images/fastq_format.png" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="alignment-essential" class="slide level2">
<h2>Alignment essential</h2>
<p>To align RNA-seq reads, two main inputs are required besides the FASTQ files:</p>
<ol type="1">
<li class="fragment">Reference Genome (FASTA)</li>
<li class="fragment">Gene Annotation (GTF)</li>
</ol>
<blockquote>
<p>These define the genomic coordinates and transcript structures needed for accurate alignment and quantification.</p>
</blockquote>
</section>
<section id="reference-genome-fasta" class="slide level2">
<h2>Reference Genome (FASTA)</h2>
<p>A FASTA file contains the nucleotide sequences of all chromosomes in the organism.</p>
<p>Used by aligners (e.g., STAR, HISAT2) to determine where each read originates.</p>
<p>Example:</p>
<p>&gt;chr1</p>
<p>ATGCGTAACTGATCGT‚Ä¶</p>
<blockquote>
<p>Must match the annotation (same genome build, e.g., GRCh38, mm10).</p>
</blockquote>
</section>
<section id="gene-annotation-file-gtf" class="slide level2">
<h2>Gene Annotation File (GTF)</h2>
<p>A GTF (Gene Transfer Format) file provides gene and transcript coordinates on the genome.</p>
<p>It tells the aligner or quantifier where exons, transcripts, and genes are located.</p>
<p>Example entry:</p>
<p>chr1 HAVANA exon 11869 12227 . + . gene_id ‚ÄúENSG00000223972‚Äù; transcript_id ‚ÄúENST00000456328‚Äù;</p>
<blockquote>
<p>Crucial for splice-aware alignment and feature counting (e.g., HTSeq, featureCounts).</p>
</blockquote>
</section>
<section id="post-alignment-bamsam" class="slide level2">
<h2>Post-alignment (BAM/SAM)</h2>
<p>After alignment, reads are stored in SAM (Sequence Alignment/Map) format ‚Äî a text-based file describing where each read maps to the reference genome. The BAM format is its binary, compressed version, faster for storage and processing.</p>
<blockquote>
<p>Contain mapping position, strand, mapping quality, and alignment flags for each read.</p>
</blockquote>
</section>
<section id="post-alignment-bamsam-1" class="slide level2">
<h2>Post-alignment (BAM/SAM)</h2>

<img data-src="https://samformat.pages.dev/images/sam_format_annotated_example.5108a0cd.jpg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="quantification-count-matrix" class="slide level2">
<h2>Quantification (count matrix)</h2>
<p>The count matrix is a table where rows = genes and columns = samples, showing how many reads align to each gene. Generated after alignment and annotation (e.g., via featureCounts or HTSeq-count).</p>
</section>
<section id="quantification-count-matrix-1" class="slide level2">
<h2>Quantification (count matrix)</h2>

<img data-src="https://www.researchgate.net/profile/Zeynab-Maghsoudi/publication/380400909/figure/fig4/AS:11431281244042660@1715797649467/An-example-of-the-content-of-a-count-matrix-file-In-this-example-the-count-matrix-file.ppm" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="other-post-alignment-analysis" class="slide level2 smaller">
<h2>Other post-alignment analysis</h2>
<p><strong>1) Transcript Assembly</strong> ‚Äì Reconstructs full-length transcripts from aligned reads to discover novel isoforms or unannotated genes. Tools: StringTie, Cufflinks.</p>
<p><strong>2) RNA Editing Detection</strong> ‚Äì Identifies base modifications such as A‚ÜíI or C‚ÜíU editing events by comparing RNA reads to the reference genome. Use case: Brain and immune tissues often show strong editing signatures.</p>
<p><strong>3) Variant Calling from RNA-seq</strong> ‚Äì Detects SNPs and small indels directly from transcribed regions, revealing expressed genetic variants. Tools: GATK, bcftools, FreeBayes.</p>
<p><strong>4) Gene Fusion Detection</strong> ‚Äì Finds chimeric transcripts produced by genomic rearrangements, common in cancer and developmental disorders. Tools: STAR-Fusion, FusionCatcher.</p>
</section>
<section id="nextflow-and-nf-core" class="slide level2 smaller">
<h2>Nextflow and nf-core</h2>
<p>Nextflow is a workflow management system designed for reproducible, scalable, and portable bioinformatics pipelines. It allows researchers to write pipelines in a simple scripting language that can be executed seamlessly on local machines, HPC clusters, or cloud environments.</p>
<p>nf-core is a community-driven platform that provides a collection of high-quality, peer-reviewed Nextflow pipelines following best practices. Each nf-core pipeline is version-controlled, well-documented, and designed for easy adaptation to new datasets.</p>
<blockquote>
<p>Together, Nextflow and nf-core form a powerful ecosystem for standardized, reproducible omics analysis.</p>
</blockquote>
</section>
<section id="rnaseq-workflow" class="slide level2">
<h2>RNAseq workflow</h2>

<img data-src="https://raw.githubusercontent.com/nf-core/rnaseq/3.21.0//docs/images/nf-core-rnaseq_metro_map_grey_animated.svg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section class="slide level2">

</section></section>
<section>
<section id="downstream-analysis" class="title-slide slide level1 center">
<h1>Downstream analysis</h1>

</section>
<section id="differential-gene-expression-dge" class="slide level2">
<h2>Differential Gene Expression (DGE)</h2>
<p><strong>Goal:</strong> Identify genes with significant expression changes between conditions (e.g., control vs.&nbsp;treated).</p>
<p><strong>Input:</strong> Normalized count matrix (e.g., from DESeq2 or edgeR).</p>
<p><strong>Output:</strong> Fold-change and adjusted p-values.</p>
<p><strong>Tools:</strong> DESeq2, edgeR, limma-voom.</p>
<p><strong>Common visualization:</strong> Volcano plot, heatmap, PCA.</p>
</section>
<section id="functional-enrichment-analysis" class="slide level2">
<h2>Functional Enrichment Analysis</h2>
<p>Goal: Discover biological processes or pathways enriched among a set of genes (or DEGs).</p>
<p>Approaches:</p>
<ol type="1">
<li class="fragment"><p>Over-representation analysis (ORA) ‚Äì tests DEGs vs.&nbsp;background.</p></li>
<li class="fragment"><p>Gene Set Enrichment Analysis (GSEA) ‚Äì uses ranked gene lists.</p></li>
</ol>
<p>Databases: GO, KEGG, Reactome, MSigDB.</p>
<p>Tools: clusterProfiler, fgsea, enrichR.</p>
</section>
<section id="proteinprotein-interaction-ppi-networks" class="slide level2">
<h2>Protein‚ÄìProtein Interaction (PPI) Networks</h2>
<p><strong>Goal:</strong> Explore how DEG-encoded proteins interact.</p>
<p><strong>Data sources:</strong> STRING, BioGRID, IntAct.</p>
<p><strong>Analysis:</strong> Identify hub proteins, clusters, or signaling modules.</p>
<p><strong>Tools:</strong> STRINGdb (R), Cytoscape, NetworkAnalyst.</p>
</section>
<section id="co-expression-network-analysis" class="slide level2">
<h2>Co-expression Network Analysis</h2>
<p><strong>Goal:</strong> Identify modules of co-expressed genes sharing biological functions.</p>
<p><strong>Approach:</strong> Correlate gene expression profiles across samples.</p>
<p><strong>Tools:</strong> WGCNA, CEMiTool, MEGENA.</p>
<p><strong>Output:</strong> Network modules ‚Üí correlated with phenotypes or traits.</p>
</section>
<section id="cell-type-deconvolution" class="slide level2">
<h2>Cell-type Deconvolution</h2>
<p><strong>Goal:</strong> Estimate cell-type proportions from bulk RNA-seq data.</p>
<p><strong>Why:</strong> Bulk samples are mixtures of multiple cell types.</p>
<p><strong>Approach:</strong> Compare to reference scRNA-seq or purified profiles.</p>
<p><strong>Tools:</strong> CIBERSORTx, MuSiC, BisqueRNA, xCell.</p>
</section>
<section id="other-downstream-analyses" class="slide level2">
<h2>Other Downstream Analyses</h2>
<p><strong>1) Alternative Splicing Analysis</strong> ‚Äî detect isoform changes (rMATS, SUPPA2).</p>
<p><strong>2) Gene Regulatory Network Inference</strong> ‚Äî identify TF‚Äìtarget relationships (SCENIC, GENIE3).</p>
<p><strong>3) Integration with Multi-Omics</strong> ‚Äî combine transcriptomics with proteomics or epigenomics (MOFA, mixOmics).</p>
<p><strong>4) Visualization Dashboards</strong> ‚Äî streamline exploration (Shiny, Streamlit, iDEP).</p>
</section>
<section id="fictional-design-request-1" class="slide level2">
<h2>Fictional design request #1</h2>
<p>A biologist have an experiment that it needs to compare KO cells vs.&nbsp;normal cells. He is not aware of which mechanism is involved but have a hint related to splicing machinery might be involved. He wants to know:</p>
<ul>
<li class="fragment">How many samples should be sequenced?</li>
<li class="fragment">How much depth should be sequenced per sample? Read length?</li>
<li class="fragment">Which library should be used?</li>
</ul>
</section>
<section id="fictional-design-request-2" class="slide level2">
<h2>Fictional design request #2</h2>
<p>A immunologist have an experiment that it needs to compare 4 conditions in 3 different time points. She doesn¬¥t want to bother with anything new or out of the shelf, just find basic inflamatory mechanism. She wants to know:</p>
<ul>
<li class="fragment">How many samples should be sequenced?</li>
<li class="fragment">How much depth should be sequenced per sample? Read length?</li>
<li class="fragment">Which library should be used?</li>
</ul>
</section>
<section id="break" class="slide level2">
<h2>Break</h2>
<p>‚òï 15 min coffee break<br>
Stretch, chat, and prepare for the next section:<br>
<strong>Application &amp; Case Study</strong></p>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="application-case-study" class="title-slide slide level1 center">
<h1>Application &amp; Case study</h1>

</section>
<section id="today-assay" class="slide level2">
<h2>Today assay</h2>
<ul>
<li class="fragment">Research for ~1h for a short-read RNAseq experiment in any research study. Make a two slides presentation (5 minutes or less) in our colective google presentations:</li>
<li class="fragment">Answer these questions:
<ul>
<li class="fragment">Which sequencing platform was used?</li>
<li class="fragment">Single-end or paired-end? Read length?</li>
<li class="fragment">Which genome reference was used? If suitable, which annotation reference?</li>
<li class="fragment">Does it follow ENCODE best principles?</li>
</ul></li>
</ul>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="questions" class="title-slide slide level1 center">
<h1>Questions</h1>

</section>
<section id="section" class="slide level2">
<h2>‚Ä¶</h2>
<ol type="A">
<li class="fragment">‚Ä¶</li>
<li class="fragment">‚Ä¶</li>
<li class="fragment">‚Ä¶</li>
<li class="fragment">‚Ä¶</li>
<li class="fragment"><strong>B) Detect RNA expression across thousands of genes simultaneously</strong></li>
</ol>
<div class="quarto-auto-generated-content">
<div class="footer footer-default">

</div>
</div>
</section></section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="day_3_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="day_3_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="day_3_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="day_3_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="day_3_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="day_3_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="day_3_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="day_3_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="day_3_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="day_3_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'fade',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        text: function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
        const config = {
          allowHTML: true,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start',
        };
        if (contentFn) {
          config.content = contentFn;
        }
        if (onTriggerFn) {
          config.onTrigger = onTriggerFn;
        }
        if (onUntriggerFn) {
          config.onUntrigger = onUntriggerFn;
        }
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>