<!DOCTYPE html>
<html lang="en"><head>
<script src="day_2_files/libs/clipboard/clipboard.min.js"></script>
<script src="day_2_files/libs/quarto-html/tabby.min.js"></script>
<script src="day_2_files/libs/quarto-html/popper.min.js"></script>
<script src="day_2_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="day_2_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="day_2_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="day_2_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="day_2_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.4.550">

  <title>Day 2 - Microarray method</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="day_2_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="day_2_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
  </style>
  <link rel="stylesheet" href="day_2_files/libs/revealjs/dist/theme/quarto.css">
  <link href="day_2_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="day_2_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="day_2_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="day_2_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    padding-bottom: 0.5rem;
    margin-bottom: 0;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Day 2 - Microarray method</h1>
  <p class="subtitle">First Genome-Wide View</p>

<div class="quarto-title-authors">
</div>

</section>
<section id="course-topics" class="slide level2">
<h2>Course Topics</h2>
<ol type="1">
<li class="fragment"><strong>Day 1:</strong> Brief Introduction + Quantitative RNA methods;</li>
<li class="fragment"><strong>Day 2: Microarray: First Genome-Wide View;</strong></li>
</ol>
</section>
<section id="yesterday-summary" class="slide level2">
<h2>Yesterday summary</h2>
<div class="cell">
<div class="cell-output-display">
<table>
<thead>
<tr class="header">
<th style="text-align: left;">Method</th>
<th style="text-align: right;">Year</th>
<th style="text-align: left;">RNA Required</th>
<th style="text-align: left;">Time</th>
<th style="text-align: left;">Throughput</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Northern Blot</td>
<td style="text-align: right;">1977</td>
<td style="text-align: left;">μg</td>
<td style="text-align: left;">2-3 days</td>
<td style="text-align: left;">1 gene</td>
</tr>
<tr class="even">
<td style="text-align: left;">RT-PCR</td>
<td style="text-align: right;">1985</td>
<td style="text-align: left;">ng</td>
<td style="text-align: left;">8 hours</td>
<td style="text-align: left;">5-10 genes</td>
</tr>
<tr class="odd">
<td style="text-align: left;">qPCR</td>
<td style="text-align: right;">1996</td>
<td style="text-align: left;">pg-ng</td>
<td style="text-align: left;">2 hours</td>
<td style="text-align: left;">96-384 genes</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="todays-goals" class="slide level2">
<h2>Today’s Goals</h2>
<p>By the end of today, you should be able to:</p>
<ol type="1">
<li class="fragment"><strong>Principles of microarray technology</strong>
<ul>
<li class="fragment">Probes, hybridization, fluorescence detection.</li>
<li class="fragment">One-color vs two-color systems</li>
</ul></li>
<li class="fragment"><strong>Normalization methods (RMA, quantile)</strong><br>
</li>
<li class="fragment"><strong>Statistical analysis concepts</strong>
<ul>
<li class="fragment">Fold-change, p-value, multiple testing.<br>
</li>
<li class="fragment">Differential expression and others.</li>
</ul></li>
</ol>
</section>
<section class="slide level2">

</section>
<section>
<section id="principles-of-microarray" class="title-slide slide level1 center">
<h1>Principles of microarray</h1>

</section>
<section id="principles-of-microarray-technology" class="slide level2">
<h2>Principles of Microarray Technology</h2>
<p>Microarrays allow <strong>simultaneous measurement of thousands of RNA transcripts</strong> by hybridizing labeled cDNA to probes fixed on a solid surface.</p>
<ul>
<li class="fragment"><strong>Probes:</strong> Short DNA sequences (~25–70 bp) immobilized on a glass slide or chip.</li>
<li class="fragment"><strong>Hybridization:</strong> Labeled sample cDNA binds to complementary probes.</li>
<li class="fragment"><strong>Detection:</strong> Fluorescent intensity at each spot corresponds to transcript abundance.</li>
</ul>
</section>
<section id="defining-probes-again" class="slide level2">
<h2>Defining probes (again)</h2>

<img data-src="https://www.researchgate.net/publication/280580151/figure/fig1/AS:651158213361673@1532259624752/DNA-microarray-with-capture-probes-The-microarray-is-made-using-photolithography-or.png" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"><p>In the context of microarray, a probe is a short, <strong>single-stranded DNA sequence immobilized on the array surface that specifically hybridizes to a complementary target RNA (or cDNA) sequence</strong>, enabling the detection and quantification of gene expression levels.</p>
</section>
<section id="microarray-workflow-overview" class="slide level2">
<h2>Microarray Workflow Overview</h2>
<ol type="1">
<li class="fragment"><strong>RNA extraction</strong> from control and test samples<br>
</li>
<li class="fragment"><strong>Reverse transcription</strong> to cDNA with fluorescent labeling<br>
</li>
<li class="fragment"><strong>Hybridization</strong> of cDNA to complementary probes on chip<br>
</li>
<li class="fragment"><strong>Washing</strong> to remove unbound cDNA<br>
</li>
<li class="fragment"><strong>Scanning</strong> to measure fluorescence<br>
</li>
<li class="fragment"><strong>Normalization and analysis</strong> of intensity data</li>
</ol>
</section>
<section id="microarray-workflow-overview-1" class="slide level2">
<h2>Microarray Workflow Overview</h2>

<img data-src="https://microbenotes.com/wp-content/uploads/2018/09/Steps-Involved-in-cDNA-based-Microarray.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="microarray-workflow-overview-2" class="slide level2">
<h2>Microarray Workflow Overview</h2>
<div style="display: flex; justify-content: center; align-items: center; height: 60vh;">
<p><iframe src="https://www.youtube.com/embed/0ATUjAxNf6U" title="Microarray technology" style="width: 80vw; height: 45vw; max-width: 800px; max-height: 400px; border: none;" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""> </iframe></p>
</div>
</section>
<section id="one-or-two-color-arrays" class="slide level2">
<h2>One or two color arrays?</h2>
<p>A major design consideration in a microarray experiment is whether to measure the expression levels from each sample on separate microarrays (one-colour array) or to compare relative expression levels between a pair of samples on a single microarray (two-colour array). The overall performance of one-colour and two-colour arrays is similar.</p>
<blockquote>
<p>Source: EBI functional genomics courses</p>
</blockquote>
</section>
<section id="one-or-two-color-arrays-1" class="slide level2">
<h2>One or two color arrays?</h2>

<img data-src="https://www.ebi.ac.uk/training/online/courses/functional-genomics-ii-common-technologies-and-data-analysis-methods/wp-content/uploads/sites/70/2020/05/Figure11_blue-1024x738.png" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="one-color-vs-two-color-systems" class="slide level2">
<h2>One-Color vs Two-Color Systems</h2>
<h3 id="one-color-single-channel">🟩 One-Color (Single Channel)</h3>
<ul>
<li class="fragment">Each sample hybridized to its own array.</li>
<li class="fragment">Common in <strong>Affymetrix</strong> and <strong>Illumina</strong> platforms.</li>
<li class="fragment">Signal intensity represents <strong>abs. expression</strong> for one sample.</li>
</ul>
<h3 id="two-color-dual-channel">🟥🟩 Two-Color (Dual Channel)</h3>
<ul>
<li class="fragment">Two samples hybridized on the <strong>same array</strong>:
<ul>
<li class="fragment">Example: Control = Cy3 (green), Treated = Cy5 (red)</li>
</ul></li>
<li class="fragment">Measures <strong>relative expression</strong> directly by color ratio.</li>
</ul>
</section>
<section id="comparison-and-trends" class="slide level2">
<h2>Comparison and trends</h2>
<table>
<colgroup>
<col style="width: 15%">
<col style="width: 23%">
<col style="width: 17%">
<col style="width: 13%">
<col style="width: 30%">
</colgroup>
<thead>
<tr class="header">
<th>Platform</th>
<th>Probes</th>
<th>Typical Design</th>
<th>Color System</th>
<th>Notes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>Affymetrix</strong></td>
<td>Short (25-mer)</td>
<td>Photolithography</td>
<td>One-color</td>
<td>Historical gold standard</td>
</tr>
<tr class="even">
<td><strong>Agilent</strong></td>
<td>Long (~60-mer)</td>
<td>Inkjet synthesis</td>
<td>Two-color</td>
<td>Popular for academic studies</td>
</tr>
<tr class="odd">
<td><strong>Illumina</strong></td>
<td>Bead-based</td>
<td>BeadArray</td>
<td>One-color</td>
<td>High reproducibility</td>
</tr>
</tbody>
</table>
</section>
<section id="affymetrix" class="slide level2">
<h2>Affymetrix</h2>

<img data-src="https://upload.wikimedia.org/wikipedia/commons/2/22/Affymetrix-microarray.jpg?20060410111635" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="agilent" class="slide level2">
<h2>Agilent</h2>

<img data-src="https://pbs.twimg.com/media/EX1DYQMXQAI6Vva.jpg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section></section>
<section>
<section id="normalization" class="title-slide slide level1 center">
<h1>Normalization</h1>

</section>
<section id="normalization-method" class="slide level2">
<h2>Normalization method</h2>
<p>Normalization is the process of <strong>adjusting microarray data to remove systematic technical variation</strong> that is not related to biological differences. Most known and used method is <strong>RMA Normalization</strong> (Robust Multi-array Average).</p>
</section>
<section id="rma-normalization" class="slide level2">
<h2>RMA Normalization</h2>
<p><strong>Purpose:</strong> To correct background noise and make expression values comparable across microarrays.</p>
<p><strong>Main Steps:</strong></p>
<ul>
<li class="fragment">Background correction – adjusts for optical noise and nonspecific signal.</li>
<li class="fragment">Quantile normalization – forces all arrays to have the same overall distribution. You can possibly use only this as a normalization method.</li>
<li class="fragment">Summarization – combines probe-level data into one expression value per gene using a median polish model.</li>
</ul>
</section>
<section id="rma-normalization-1" class="slide level2">
<h2>RMA Normalization</h2>
<p><strong>Advantages:</strong></p>
<ul>
<li class="fragment">Reduces array-to-array variability.</li>
<li class="fragment">Improves reproducibility in downstream differential expression analysis.</li>
<li class="fragment">Implemented in tools like <strong>affy</strong> and <strong>oligo</strong> (R/Bioconductor).</li>
</ul>
</section>
<section id="quantile-normalization" class="slide level2">
<h2>Quantile Normalization</h2>

<img data-src="https://www.researchgate.net/profile/Saroj-Mohapatra/publication/47371361/figure/fig1/AS:306044065599492@1449977999233/Quantile-normalization-imposes-the-same-empirical-distribution-of-intensities-on-each.png" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section class="slide level2">

</section></section>
<section>
<section id="statistical-analysis" class="title-slide slide level1 center">
<h1>Statistical analysis</h1>

</section>
<section id="fold-change-in-microarray-experiments" class="slide level2">
<h2>Fold Change in Microarray Experiments</h2>
<p>Fold change quantifies the <strong>magnitude of gene expression difference</strong> between two conditions (e.g., treated vs.&nbsp;control).</p>
<p>Fold Change = Mean Expression (treated) / Mean Expression (control)</p>
<p>To symmetrize up- and down-regulation: log2(Fold Change) → +1 = 2× up, -1 = 2× down</p>
<blockquote>
<p>Fold change alone does not indicate statistical significance — it measures effect size only.</p>
</blockquote>
</section>
<section id="p-value-and-multiple-testing" class="slide level2">
<h2>p-Value and Multiple Testing</h2>
<p><strong>p-value:</strong> probability of observing a difference as extreme as the measured one, assuming no true difference (null hypothesis). In microarrays, we test thousands of genes simultaneously → <strong>many false positives by chance.</strong></p>
<p>Need multiple testing correction to control for these errors:</p>
<ul>
<li class="fragment">Bonferroni correction: very strict, divides α by number of tests.</li>
<li class="fragment"><strong>Benjamini–Hochberg (FDR):</strong> controls expected false discovery rate; widely used in microarray and RNA-seq.</li>
</ul>
</section>
<section id="limma-in-microarray-analysis" class="slide level2">
<h2>limma in Microarray Analysis</h2>
<p>limma (Linear Models for Microarray Data) is one of the most widely used R packages for analyzing microarray (and RNA-seq) data. It provides a robust framework for identifying <strong>differentially expressed genes</strong> using linear models and empirical Bayes statistics.</p>
</section>
<section id="volcano-plot" class="slide level2">
<h2>Volcano plot</h2>

<img data-src="https://www.ncbi.nlm.nih.gov/geo/img/geo2r_volcano.png" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="venn-diagram" class="slide level2">
<h2>Venn Diagram</h2>

<img data-src="https://www.ncbi.nlm.nih.gov/geo/img/geo2r_vennDiagram.png" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="heatmap" class="slide level2">
<h2>Heatmap</h2>

<img data-src="https://s3-eu-west-1.amazonaws.com/ppreviews-plos-725668748/2406979/preview.jpg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="nanostring-2003" class="slide level2">
<h2>Nanostring (2003)</h2>

<img data-src="https://media.bizj.us/view/img/12485846/nanostring-technologies-biotech-in-seattle-april-2023*900xx3553-1997-23-0.jpg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="nanostring" class="slide level2">
<h2>Nanostring</h2>
<p><strong>Microarray:</strong> Relies on hybridization where labeled sample nucleic acids bind to complementary sequences on a solid surface. The amount of binding is then measured by fluorescence, which is proportional to the amount of target RNA. This method requires amplification steps.</p>
<p><strong>NanoString:</strong> Uses a direct-counting, amplification-free method. It attaches fluorescent “reporter” probes to the target nucleic acids, which are then counted directly, providing a more precise count of the original molecules.</p>
<blockquote>
<p>Both of them use same differential gene expression methods.</p>
</blockquote>
</section>
<section id="nanostring-is-bankruptcy" class="slide level2">
<h2>Nanostring is bankruptcy</h2>

<img data-src="https://www.wilsongunn.com/wp-content/uploads/2024/01/Nanostring-vs-10x-Genomics-LinkedIn.png" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"><p><strong>Patent infringement lawsuit:</strong> In November 2023, a federal jury ordered NanoString to pay competitor 10x Genomics $31 million in damages for infringing on its patents with NanoString’s GeoMx Digital Spatial Profiler. This loss created significant financial strain for the company.</p>
</section>
<section id="case-1-fu-et-al-2015" class="slide level2">
<h2>Case 1 (Fu et al, 2015)</h2>

<img data-src="https://storage.googleapis.com/plos-corpus-prod/10.1371/journal.pone.0063249/1/pone.0063249.g002.PNG_L?X-Goog-Algorithm=GOOG4-RSA-SHA256&amp;X-Goog-Credential=wombat-sa%40plos-prod.iam.gserviceaccount.com%2F20251028%2Fauto%2Fstorage%2Fgoog4_request&amp;X-Goog-Date=20251028T032848Z&amp;X-Goog-Expires=86400&amp;X-Goog-SignedHeaders=host&amp;X-Goog-Signature=0cab56ed512f3f93f070ce13a9f1096a469fc1d02c32b813fcd85ba673466d0d233ca4ce2c80de54b3f4c33e896be4994813a80a24b05b6a9c5761867f3573a9d4f1614be0ab3837a625ea341b68054e64f2347264f824734b97683177dcb09097594dbfa3742162b53d0b8c8d89001302b4b796db9aa3f05c16a064d18c1e6bf3ed376d59110dc7ce9f4160d9d81bbbfed07f447aac56a55fe859c454977682e95adaa99c92b7ddf26163d33f2700d9df299e45161af36198c8186829302299a2f4ba0a4464925b7156e9d65470e8e0914763ac216d5c6d916a720e126b20ec95b6fced1ceeac164c8f43eb39cbfade6938602e67724f682fe76f12a8b91fc4" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"><p><small> <strong>Figure 2.</strong> Heatmap representing the expressions of 59 genes with significant different expression in CUP compared with other cancer types or normal lymph nodes. Genes are indicated on the right. The colored bar above the heatmap represents the different cancer types, and the legend key is on the left. On the heatmap, red represents up-regulated genes and green represents down-regulated genes, relative to the expression levels in normal lymph nodes, with the scale shown in the upper left corner. </small></p>
</section>
<section id="break" class="slide level2">
<h2>Break</h2>
<p>☕ 15 min coffee break<br>
Stretch, chat, and prepare for the next section:<br>
<strong>Application &amp; Case Study</strong></p>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="application-case-study" class="title-slide slide level1 center">
<h1>Application &amp; Case study</h1>

</section>
<section id="today-assay" class="slide level2">
<h2>Today assay</h2>
<ul>
<li class="fragment">Research for ~1h for a microarray experiment in any research study. Make a two slides presentation (5 minutes or less) in our colective google presentations: <small>https://docs.google.com/presentation/d/1Qfn0hTdl1h5wGEnfWhgRKN1QMA4bxfer6mUtIFANL68/edit?usp=sharing</small></li>
<li class="fragment">Answer these questions:
<ul>
<li class="fragment">Probe platform used? One or two color method?</li>
<li class="fragment">Normalization method used? Highest and lowest fold-change.</li>
<li class="fragment">How many differentially expressed genes were found?</li>
</ul></li>
</ul>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="questions" class="title-slide slide level1 center">
<h1>Questions</h1>

</section>
<section id="section" class="slide level2">
<h2>…</h2>
<ol type="A">
<li class="fragment">…</li>
<li class="fragment">…</li>
<li class="fragment">…</li>
<li class="fragment">…</li>
</ol>
<p><strong>A) …</strong></p>
<div class="quarto-auto-generated-content">
<div class="footer footer-default">

</div>
</div>
</section></section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="day_2_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="day_2_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="day_2_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="day_2_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="day_2_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="day_2_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="day_2_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="day_2_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="day_2_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="day_2_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'fade',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        text: function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
        const config = {
          allowHTML: true,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start',
        };
        if (contentFn) {
          config.content = contentFn;
        }
        if (onTriggerFn) {
          config.onTrigger = onTriggerFn;
        }
        if (onUntriggerFn) {
          config.onUntrigger = onUntriggerFn;
        }
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>